src/components/modals/TradePreviewModalGP.vue

```diff
@@ -435,6 +471,41 @@ export default defineComponent({
       emit('close');
     }
 
+    function cofirmPriceUpdate() {
+      priceUpdated.value = false;
+      priceUpdateAccepted.value = true;
+      lastQuote.value = props.trading.getQuote();
+    }
+
+    function handlePriceUpdate() {
+      if (lastQuote.value != null) {
+        const newQuote = props.trading.getQuote();
+
+        if (props.trading.exactIn.value) {
+          priceUpdated.value = bnum(lastQuote.value.minimumOutAmount)
+            .minus(newQuote.minimumOutAmount)
+            .abs()
+            .div(lastQuote.value.minimumOutAmount)
+            .gt(PRICE_UPDATE_THRESHOLD);
+        } else {
+          priceUpdated.value = bnum(lastQuote.value.maximumInAmount)
+            .minus(newQuote.maximumInAmount)
+            .abs()
+            .div(lastQuote.value.maximumInAmount)
+            .gt(PRICE_UPDATE_THRESHOLD);
+        }
+
+        if (priceUpdated.value) {
+          priceUpdateAccepted.value = false;
+        }
+      }
+    }
+
+    // WATCHERS
+    watch(blockNumber, () => {
+      handlePriceUpdate();
```

Comment:
Check needs to happen every block (that's when Gnosis/Balancer price updates kick in)